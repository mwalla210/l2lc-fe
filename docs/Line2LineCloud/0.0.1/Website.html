<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Website - Documentation</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc.css">
</head>
<body>

<input type="checkbox" id="nav-trigger" class="nav-trigger" />
<label for="nav-trigger" class="navicon-button x">
  <div class="navicon"></div>
</label>

<label for="nav-trigger" class="overlay"></label>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="CustomerModel.html">CustomerModel</a><ul class='methods'><li data-type='method'><a href="CustomerModel.html#edit">edit</a></li><li data-type='method'><a href="CustomerModel.html#getformattedBillAddress">getformattedBillAddress</a></li><li data-type='method'><a href="CustomerModel.html#getformattedShipAddress">getformattedShipAddress</a></li><li data-type='method'><a href="CustomerModel.html#getProjects">getProjects</a></li></ul></li><li><a href="EmployeeModel.html">EmployeeModel</a><ul class='methods'><li data-type='method'><a href="EmployeeModel.html#activate">activate</a></li><li data-type='method'><a href="EmployeeModel.html#changeName">changeName</a></li><li data-type='method'><a href="EmployeeModel.html#deactivate">deactivate</a></li></ul></li><li><a href="Page.html">Page</a><ul class='methods'><li data-type='method'><a href="Page.html#customerSelect">customerSelect</a></li></ul></li><li><a href="ProjectModel.html">ProjectModel</a><ul class='methods'><li data-type='method'><a href="ProjectModel.html#addRework">addRework</a></li><li data-type='method'><a href="ProjectModel.html#addTask">addTask</a></li><li data-type='method'><a href="ProjectModel.html#addTimeEntry">addTimeEntry</a></li><li data-type='method'><a href="ProjectModel.html#changeCustomer">changeCustomer</a></li><li data-type='method'><a href="ProjectModel.html#changeHold">changeHold</a></li><li data-type='method'><a href="ProjectModel.html#customerID">customerID</a></li><li data-type='method'><a href="ProjectModel.html#delete">delete</a></li><li data-type='method'><a href="ProjectModel.html#edit">edit</a></li><li data-type='method'><a href="ProjectModel.html#finish">finish</a></li><li data-type='method'><a href="ProjectModel.html#getDefaultTaskList">getDefaultTaskList</a></li><li data-type='method'><a href="ProjectModel.html#moveTask">moveTask</a></li><li data-type='method'><a href="ProjectModel.html#removeTask">removeTask</a></li><li data-type='method'><a href="ProjectModel.html#status">status</a></li><li data-type='method'><a href="ProjectModel.html#timeSpent">timeSpent</a></li><li data-type='method'><a href="ProjectModel.html#toggleTask">toggleTask</a></li></ul></li><li><a href="TableModel.html">TableModel</a><ul class='methods'><li data-type='method'><a href="TableModel.html#dataFetch">dataFetch</a></li></ul></li><li><a href="TaskModel.html">TaskModel</a><ul class='methods'><li data-type='method'><a href="TaskModel.html#toggle">toggle</a></li></ul></li><li><a href="UserModel.html">UserModel</a><ul class='methods'><li data-type='method'><a href="UserModel.html#changePassword">changePassword</a></li><li data-type='method'><a href="UserModel.html#login">login</a></li><li data-type='method'><a href="UserModel.html#logout">logout</a></li><li data-type='method'><a href="UserModel.html#toggleAdmin">toggleAdmin</a></li></ul></li><li><a href="Website.html">Website</a><ul class='methods'><li data-type='method'><a href="Website.html#createCustomer">createCustomer</a></li><li data-type='method'><a href="Website.html#createEmployee">createEmployee</a></li><li data-type='method'><a href="Website.html#createProject">createProject</a></li><li data-type='method'><a href="Website.html#generateBarcode">generateBarcode</a></li><li data-type='method'><a href="Website.html#getAccounts">getAccounts</a></li><li data-type='method'><a href="Website.html#getCustomers">getCustomers</a></li><li data-type='method'><a href="Website.html#getEmployees">getEmployees</a></li><li data-type='method'><a href="Website.html#getProject">getProject</a></li><li data-type='method'><a href="Website.html#getProjects">getProjects</a></li></ul></li></ul>
</nav>

<div id="main">
    
    <h1 class="page-title">Website</h1>
    

    




<section>

<header>
    
        <h2>
        Website
        </h2>
        
            <div class="class-description"><p>Main MobX store for website</p></div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    <h4 class="name" id="Website">@store <span class="type-signature"></span>new Website<span class="signature">()</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line4">line 4</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Add Analytic models & reference</li>
        </ul>
    </dd>
    
</dl>



    <h5 class="subsection-title">Properties:</h5>

    

<table class="props">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        
        <th>Attributes</th>
        

        
        <th>Default</th>
        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>currentProject</code></td>
            

            <td class="type">
            
                
<span class="param-type">Project</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Current Project in state, or last focused Project.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>currentCustomer</code></td>
            

            <td class="type">
            
                
<span class="param-type">Customer</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Current Customer in state, or last focused Customer.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>currentUser</code></td>
            

            <td class="type">
            
                
<span class="param-type">User</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Current User in state, or last focused User.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>openProjects</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Project></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Current array of Project(s) in state, if any, that are not marked as closed in database.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>closedProjects</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Project></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Current array of Project(s) in state, if any, that are marked as closed in database.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>accounts</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;User></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Current array of User(s) in state, if any.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>employees</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Employee></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Current array of Employee(s) in state, if any.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>customers</code></td>
            

            <td class="type">
            
                
<span class="param-type">Array.&lt;Customer></span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>[]</code>
                
                </td>
            

            <td class="description last"><p>Current array of Customer(s) in state, if any.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>ccAnalytic</code></td>
            

            <td class="type">
            
                
<span class="param-type">Analytic</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Analytic model in state. May be default model if not fetched yet.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>eAnalytic</code></td>
            

            <td class="type">
            
                
<span class="param-type">Analytic</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Analytic model in state. May be default model if not fetched yet.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>paAnalytic</code></td>
            

            <td class="type">
            
                
<span class="param-type">Analytic</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Analytic model in state. May be default model if not fetched yet.</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>jtAnalytic</code></td>
            

            <td class="type">
            
                
<span class="param-type">Analytic</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                    &lt;observable><br>
                
                </td>
            

            
                <td class="default">
                
                    <code>null</code>
                
                </td>
            

            <td class="description last"><p>Analytic model in state. May be default model if not fetched yet.</p></td>
        </tr>

    
    </tbody>
</table>





























    
    </div>

    

    

    

     

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    <h4 class="name" id="createCustomer">@action <span class="type-signature"></span>createCustomer<span class="signature">(customer)</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line126">line 126</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Sends the formatted Customer in POST to API to add entry to database; sets this.currentCustomer</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>customer</code></td>
            

            <td class="type">
            
                
<span class="param-type">Customer</span>


            
            </td>

            

            

            <td class="description last"><p>Finalized Customer to create in database</p></td>
        </tr>

    
    </tbody>
</table>















        
            

    

    <h4 class="name" id="createEmployee">@action <span class="type-signature"></span>createEmployee<span class="signature">(employee)</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line138">line 138</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Sends the formatted Employee in POST to API to add entry to database; updates this.employees</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>employee</code></td>
            

            <td class="type">
            
                
<span class="param-type">Employee</span>


            
            </td>

            

            

            <td class="description last"><p>Finalized Employee to create in database</p></td>
        </tr>

    
    </tbody>
</table>















        
            

    

    <h4 class="name" id="createProject">@action <span class="type-signature"></span>createProject<span class="signature">(project)</span><span class="type-signature"></span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line114">line 114</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Sends the formatted Project in POST to API to add entry to database; sets this.currentProject</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>project</code></td>
            

            <td class="type">
            
                
<span class="param-type">Project</span>


            
            </td>

            

            

            <td class="description last"><p>Finalized Project to create in database</p></td>
        </tr>

    
    </tbody>
</table>















        
            

    

    <h4 class="name" id="generateBarcode"><span class="type-signature"></span>generateBarcode<span class="signature">(id)</span><span class="type-signature"> &rarr; {Object}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line153">line 153</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>TBD</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">Number</span>


            
            </td>

            

            

            <td class="description last"><p>Barcode ID</p></td>
        </tr>

    
    </tbody>
</table>














<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Object</span>


    </dd>
</dl>

    

        
            

    

    <h4 class="name" id="getAccounts">@action <span class="type-signature"></span>getAccounts<span class="signature">()</span><span class="type-signature"> &rarr; {Promise}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line73">line 73</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Instantiates returned collection as User(s) and sets this.accounts</p>
</div>























<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Promise</span>


    </dd>
</dl>

    

        
            

    

    <h4 class="name" id="getCustomers">@action <span class="type-signature"></span>getCustomers<span class="signature">()</span><span class="type-signature"> &rarr; {Promise}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line99">line 99</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Instantiates returned collection as Customer(s) and sets this.customers</p>
</div>























<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Promise</span>


    </dd>
</dl>

    

        
            

    

    <h4 class="name" id="getEmployees">@action <span class="type-signature"></span>getEmployees<span class="signature">()</span><span class="type-signature"> &rarr; {Promise}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line86">line 86</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Instantiates returned collection as Employee(s) and sets this.employees</p>
</div>























<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Promise</span>


    </dd>
</dl>

    

        
            

    

    <h4 class="name" id="getProject">@action <span class="type-signature"></span>getProject<span class="signature">(id)</span><span class="type-signature"> &rarr; {Promise}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line45">line 45</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Sets this.currentProject as Project instantiated returned data</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>id</code></td>
            

            <td class="type">
            
                
<span class="param-type">Number</span>


            
            </td>

            

            

            <td class="description last"><p>Database ID of project to fetch</p></td>
        </tr>

    
    </tbody>
</table>














<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Promise</span>


    </dd>
</dl>

    

        
            

    

    <h4 class="name" id="getProjects">@action <span class="type-signature"></span>getProjects<span class="signature">(open<span class="signature-attributes">opt</span>)</span><span class="type-signature"> &rarr; {Promise}</span></h4>

    




<dl class="details">

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="store_website.js.html">store/website.js</a>, <a href="store_website.js.html#line59">line 59</a>
    </li></ul></dd>
    

    

    

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-todo">To Do:</dt>
    <dd class="tag-todo">
        <ul>
            <li>Implement function</li>
        </ul>
    </dd>
    
</dl>





<div class="description">
    <p>Instantiates returned collection as Project(s) and sets this.openProjects or this.closedProjects, dependent on input</p>
</div>











    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        
        <th>Attributes</th>
        

        
        <th>Default</th>
        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>open</code></td>
            

            <td class="type">
            
                
<span class="param-type">Boolean</span>


            
            </td>

            
                <td class="attributes">
                
                    &lt;optional><br>
                

                

                
                </td>
            

            
                <td class="default">
                
                    <code>true</code>
                
                </td>
            

            <td class="description last"><p>Fetch either open or closed projects</p></td>
        </tr>

    
    </tbody>
</table>














<h5>Returns:</h5>

        


<dl class="param-type">
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Promise</span>


    </dd>
</dl>

    

        
    

    

    
</article>

</section>




</div>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Jan 27 2018 17:06:44 GMT-0500 (EST) using the <a href="https://github.com/clenemt/docdash">docdash</a> theme.
</footer>

<script>prettyPrint();</script>
<script src="scripts/linenumber.js"></script>
</body>
</html>