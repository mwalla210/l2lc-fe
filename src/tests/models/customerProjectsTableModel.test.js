import CustomerProjectsTableModel from '../../models/customerProjectsTableModel'
import renderer from 'react-test-renderer'

jest.mock('../../components/projectStatusFilter')
jest.mock('../../api', () => {
  return {
    fetchCustomerProjects: jest.fn()
  }
})
jest.mock('../../store/website', () => {
  return {
    setCustomerTable: jest.fn(),
    currentProject: {
      changeCustomerTable: jest.fn()
    },
    currentUser: {
      admin: false
    },
    currentCustomer:{
      id: 1,
    },
    setProject: jest.fn().mockReturnValue(Promise.resolve()),
  }
})
import Website from '../../store/website'

describe('CustomerProjectsTableModel', () => {
  xit('Tests constructor', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn())
    expect(customerTable).toHaveProperty('columns')
    customerTable.fetchFn()
    expect(customerTable.columns[1].accessor({dateCreated: new Date('December 17, 1995 03:24:00')})).toEqual(expect.stringContaining('Dec'))
    expect(customerTable.columns[5].filterMethod({value:{length: 0}},[]))
    expect(customerTable.columns[5].filterMethod({value:{length: 9, includes: jest.fn()}},[]))
})

  it('Tests clickHandler with info', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customerTable.infoClickNav.mock.calls.length).toBe(0)
    customerTable.clickHandler(1,'info')
    expect(customerTable.infoClickNav.mock.calls.length).toBe(1)
  })

  it('Tests clickHandler with edit', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customerTable.editClickNav.mock.calls.length).toBe(0)
    customerTable.clickHandler(1,'edit')
    expect(customerTable.editClickNav.mock.calls.length).toBe(1)
  })

  it('Tests clickHandler with delete', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customerTable.clickHandler(1,'delete')
    //else case, if it ran then it ran
  })

  it('Tests clickHandler with no input', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customerTable.clickHandler(1,'null')
    //else case, if it ran then it ran
  })

  it('Tests toggleDropdown', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    customerTable.toggleDropdown()
    expect(customerTable.filterDD).toEqual(true)
  })

  xit('Tests Actions column', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    expect(customerTable.columns[6].getProps()).toEqual({className: 'center',style: {paddingTop: '0px',paddingBottom: '0px'}})
    const component = renderer.create(customerTable.columns[6].Cell({original:{toggleAdmin: jest.fn(), status: 'Completed'}}))
    let tree2 = component.toJSON()
    expect(tree2).toMatchSnapshot()
    //component.root.findByType(Switch).props.onClick()
  })

  xit('Tests Actions column', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    Website.currentUser.admin = true
    expect(customerTable.columns[6].getProps()).toEqual({className: 'center',style: {paddingTop: '0px',paddingBottom: '0px'}})
    const component = renderer.create(customerTable.columns[6].Cell({original:{toggleAdmin: jest.fn(), status: 'Completed'}}))
    let tree2 = component.toJSON()
    expect(tree2).toMatchSnapshot()
    //component.root.findByType(Switch).props.onClick()
  })

  xit('Tests Status Cell', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
    const component = renderer.create(customerTable.columns[5].Cell({}))
    let tree2 = component.toJSON()
    expect(tree2).toMatchSnapshot()
  })

  xit('Tests Status Filter', () => {
    let customerTable = new CustomerProjectsTableModel(jest.fn(),jest.fn(),jest.fn(),jest.fn())
      expect(typeof customerTable.columns[5].Filter({filter:{},onChange:jest.fn()})).toBe('object')
  })

})
