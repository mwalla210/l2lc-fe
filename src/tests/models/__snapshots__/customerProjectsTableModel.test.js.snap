// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerProjectsTableModel Tests Actions column 1`] = `
<span>
  <span>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="info"
        src="../../style/open-iconic-master/svg/info.svg"
        style={undefined}
      />
    </button>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="pencil"
        src="../../style/open-iconic-master/svg/pencil.svg"
        style={
          Object {
            "marginLeft": "2px",
          }
        }
      />
    </button>
  </span>
</span>
`;

exports[`CustomerProjectsTableModel Tests Actions column 2`] = `
<span>
  <span>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="info"
        src="../../style/open-iconic-master/svg/info.svg"
        style={undefined}
      />
    </button>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary disabled"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="pencil"
        src="../../style/open-iconic-master/svg/pencil.svg"
        style={
          Object {
            "marginLeft": "2px",
          }
        }
      />
    </button>
    <button
      aria-label="Left Align"
      className="btn-circle btn btn-outline-secondary"
      disabled={undefined}
      onClick={[Function]}
      type="button"
    >
      <img
        alt="trash"
        src="../../style/open-iconic-master/svg/trash.svg"
        style={
          Object {
            "marginLeft": "2px",
          }
        }
      />
    </button>
  </span>
</span>
`;

exports[`CustomerProjectsTableModel Tests Status Cell 1`] = `
<span>
  <span
    style={
      Object {
        "color": "#57d500",
        "transition": "all .3s ease",
      }
    }
  >
    ‚óè
  </span>
   undefined
</span>
`;
